src/main/template/presto.yaml

```diff
@@ -0,0 +1,565 @@
+            patch_node_properties:
+              command: |
+                export INSTANCE_ID=$(curl http://169.254.169.254/latest/meta-data/instance-id)
+                echo "node.id=${INSTANCE_ID}" >> /opt/presto/presto-server/etc/node.properties
+        start_presto:
+          commands:
+            start_presto:
+              command: |
+                #!/bin/bash -xe
+                /opt/presto/presto-server/bin/launcher start
+
+  coordinatorAutoScalingGroup:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    Properties:
+      LaunchConfigurationName: !Ref coordinatorLaunchConfiguration
+      TargetGroupARNs:
+        - !Ref coordinatorTargetGroup
+      MaxSize: 2
+      MinSize: 1
+      DesiredCapacity: 1
+      HealthCheckGracePeriod: 240
+      HealthCheckType: ELB
+      VPCZoneIdentifier:
+        - Fn::Select:
+          - 0
+          - Fn::Split:
+            - ","
+            - Fn::ImportValue:
+                !Sub "${vpcStackName}-private-subnets"
+        - Fn::Select:
+          - 1
+          - Fn::Split:
+            - ","
+            - Fn::ImportValue:
+                !Sub "${vpcStackName}-private-subnets"
+        - Fn::Select:
+          - 2
+          - Fn::Split:
+            - ","
+            - Fn::ImportValue:
+                !Sub "${vpcStackName}-private-subnets"
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} Coordinator
+          PropagateAtLaunch: true
+    CreationPolicy:
+      ResourceSignal:
+        Timeout: PT5M
+
+  workerAutoScalingGroup:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    Properties:
+      LaunchConfigurationName: !Ref workerLaunchConfiguration
+      TargetGroupARNs:
+        - !Ref workerTargetGroup
+      MaxSize: 20
+      MinSize: 1
+      DesiredCapacity: 1
+      HealthCheckGracePeriod: 240
+      HealthCheckType: ELB
+      VPCZoneIdentifier:
+        - Fn::Select:
+          - 0
+          - Fn::Split:
+            - ","
+            - Fn::ImportValue:
+                !Sub "${vpcStackName}-private-subnets"
+        - Fn::Select:
+          - 1
+          - Fn::Split:
+            - ","
+            - Fn::ImportValue:
+                !Sub "${vpcStackName}-private-subnets"
+        - Fn::Select:
+          - 2
+          - Fn::Split:
+            - ","
+            - Fn::ImportValue:
+                !Sub "${vpcStackName}-private-subnets"
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} Worker
+          PropagateAtLaunch: true
+    CreationPolicy:
+      ResourceSignal:
+        Timeout: PT5M
+
+Outputs:
+  SecurityGroup:
+    Description: The presto security group, so that other security groups can allow ingress from themselves to it.
+    Value: !Ref securityGroup
+    Export:
+      Name: !Sub ${AWS::StackName}-security-group
+
+  DnsName:
+    Description: The DNS name under which presto is reachable.
+    Value: !Ref securityGroup
```

Comment:
This is wrong. I think this is meant to be (I'm not sure my syntax around `Fn::ImportValue` is correct, though, I really need to learn yaml now)
```yaml
  DnsName:
    Description: The DNS name under which presto is reachable.
    Value:
      Fn::ImportValue:
        !Sub "${privateLoadBalancerStackName}-private-lb-dns-name"
    Export:
        Name: !Sub ${AWS::StackName}-dns-name
```

Reply:
Wow. I messed up my copypasta hard.